### DISCUSSION TITLE
Fetch share test token

### DISCUSSION THREAD

**Discussion by fturibak** [ludeknovy/jtl-reporter#189] [Q&A]
Is there a possibility to fetch automatically generated test tokens via Rest API ?

Once the test are finished I would like to send e-mail with links to test results ( via CI tool )

Thanks

**Comment by ludeknovy** [comment]
Yes, you can enable the share token to be generated upon test run upload in scenario settings.

![Screenshot 2022-11-11 at 8 25 50](https://user-images.githubusercontent.com/13610612/201288451-f909735e-a5dc-4a4c-93f7-f3c7b89d8ce3.png)


![Screenshot 2022-11-11 at 8 27 41](https://user-images.githubusercontent.com/13610612/201288674-d1963b91-ba32-495d-8c53-cb7d30ede58a.png)

The url format:
http://<host>/project/test/scenario/<scenario-name>/item/<id>?token=<token>


Please note that the token will only work for a given test run.

**Reply by fturibak** [reply]
Yes. I'm aware of this flag.

But is there a possibility to fetch **latest** generated token for the given scenario ?

Thanks

**Reply by ludeknovy** [reply]
No, there is no such an option. Btw the token is valid only for one specific test run.

**Reply by fturibak** [reply]
Too bad :-(  

It's exactly what I would need; fetch somehow token for the only one specific test run ( latest ); for the list of scenarios and send those links to mailing group.
Exposing some kind of API or integrating this into existing one would be a game changer for me :-)
